java/libphonenumber/src/com/google/i18n/phonenumbers/SingleFileMetadataSourceImpl.java

```diff
@@ -104,14 +105,14 @@ void loadMetadataFromFile() {
     try {
       in = new ObjectInputStream(source);
       PhoneMetadataCollection metadataCollection = loadMetadataAndCloseInput(in);
-      List<PhoneMetadata> metadataList = metadataCollection.getMetadataList();
-      if (metadataList.isEmpty()) {
+      PhoneMetadata[] metadataList = metadataCollection.metadata;
+      if (metadataList.length == 0) {
```

Comment:
I think it's fine. Conceptually, it's still a list, just implemented as an array. The array object is just called "metadata" in the nanoproto, which is confusing.